Bump March Hare to 2.21.0

michaelklishin opened the pull request:
Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

See [release notes](https://github.com/ruby-amqp/march_hare/blob/2.x-stable/ChangeLog.md#changes-between-2200-and-2210-november-24th-2016).

michaelklishin committed:
Bump March Hare to 2.21.0

See [release notes](https://github.com/ruby-amqp/march_hare/blob/2.x-stable/ChangeLog.md#changes-between-2200-and-2210-november-24th-2016).

andrewvc commented:
Glad to finally move off the milestone version :)

andrewvc commented:
@michaelklishin Mind bumping the gemspec and leaving a Changelog entry before I merge it?

michaelklishin commented:
@andrewvc if by "bumping the gemspec" you mean bumping the version, I intentionally never do that in pull requests. It's up to the maintainer to decide when to do it and what the version should be. I'm happy to add a change log entry, will do in a bit.

michaelklishin committed:
Update change log per maintainer request

I apologize for having to bump the version number. It's not
up to contributors like me to decide what version will be
the next one.

michaelklishin commented:
@andrewvc done. Feel free to alter the version I used in `CHANGELOG` ;)

michaelklishin commented:
@andrewvc not sure where'd be the appropriate place to ask this, so: I have a March Hare 3.0 release coming soon, which won't introduce any breaking API changes (at least none are intentional) and simply bump the version because it will depend on RabbitMQ Java client 4.0 (again, no breaking changes but a few interface additions and release independence from RabbitMQ server). Should I introduce a follow-up PR after you cut a release? Should the version of this repo go up to 4.3 or 5.0 then?

andrewvc commented:
@michaelklishin thanks! I'll get this merged as soon as we unbork the logstash gem this depends on. (that's causing travis to fail ATM)

WRT MarchHare 3.0, thanks for the heads up. If there are no user-facing changes then I'm fine just bumping this by a minor version.

May I ask what the reason is for bumping the Java Client to 4.0? I assume it's not semver if there are no user-facing changes.

michaelklishin commented:
@andrewvc there are user facing changes in Java client 4.0, although they are not major or even breaking for some. The main reason is to indicate that client releases are now decoupled from RabbitMQ server ones (`3.6.x` releases will continue but there will be no Java client `3.7.0`). We did the same thing for .NET.

You would be surprised what kind of changes some assume to require a major version bump. .NET client is rapidly approaching a 5.0 with fairly minor additions (and no breaking changes for most) :)

andrewvc commented:
@michaelklishin in that case a minor version bump is fine for that :) Thanks for the heads up!
